hack/feature-deploy.sh

```diff
@@ -2,11 +2,47 @@
 
 set -e
 
+if [ "$FEATURES_ENVIRONMENT" == "" ]; then
+	echo "[ERROR]: No FEATURES_ENVIRONMENT provided"
+	exit 1
+fi
+
+if [ "$FEATURES" == "" ]; then
+	echo "[ERROR]: No FEATURES provided"
+	exit 1
+fi
+
 # expect oc to be in PATH by default
-OC_TOOL="${OC_TOOL:-oc}"
+export OC_TOOL="${OC_TOOL:-oc}"
 
-for f in $FEATURES; do
-	echo "TODO add logic to implement feature '$f' for environment '$FEATURES_ENVIRONMENT'"
-done
+# expect kustomize to be in PATH by default
+KUSTOMIZE="${KUSTOMIZE:-kustomize}"
+
+# Label 1 worker node
+echo "[INFO]:labeling 1 worker node with worker-rt"
+node=$(${OC_TOOL} get nodes --selector='node-role.kubernetes.io/worker' -o name | head -1)
+${OC_TOOL} label --overwrite=true $node node-role.kubernetes.io/worker-rt=""
+
+# Override the image name when this is invoked from openshift ci
+# Not ideal, but kustomize does not support env vars directly :/
+if [ -n "${OPENSHIFT_BUILD_NAMESPACE}" ]; then
+        echo "[INFO]: Openshift CI detected, deploying using image $FULL_REGISTRY_IMAGE"
+        FULL_REGISTRY_IMAGE="registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:performance-addon-operators-registry"
+        cp feature-configs/e2e-gcp/performance-operator/operator_catalogsource.patch.yaml.in feature-configs/e2e-gcp/performance-operator/operator_catalogsource.patch.yaml
```

Comment:
ha, wow. that's kind of annoying. There's really no reasonable way around this using variables at build time that I can see. The approach you've done here is about the best we can do if we don't know the image name up front.

They have ways of transforming images for containers in pods, deployments, etc... but not for this custom catalog source resource.

Reply:
yes this is very annoying

> They have ways of transforming images for containers in pods, deployments

if you mean `kustomize edit`, that also just modifying manifests as we do here, so not much better.

But maybe I have a less uglier approach than this: using `${OPENSHIFT_BUILD_NAMESPACE}` in the e2e manifest patch, and piping it through `envsubst` before `oc apply`. Will give it a try.

Reply:
wait...
We don't need to handle ${OPENSHIFT_BUILD_NAMESPACE} at all. We don't build the image here, we want to test an existing one :)
So due to lack of an existing upstream image I use my own here as well now, until we have official images available.

Reply:
we still need the ability to pass in an image via an ENV VAR during `make feature-deploy` though.

Reply:
can you elaborate for what usecase we still need env vars? I'm asking because if we don't need to support them, we could remove download and usage of `kustomize` and use `oc -k ...`. But that would hinder usage of `envsubst` or similar.

Reply:
what other options do we have for providing an image override for an image in a private repo that we don't want ot link to in a public repo?

Reply:
good usecase indeed. I'm not aware of any other option to do that.
So I'll leave the `kustomize | oc` apply flow and don't replace it with `oc -k`, so we can simply put `envsubst` in between.
fyi @MarSik